### PR TITLE
Console - application general migration - second part

### PR DESCRIPTION
**Pull request by jansobczynski**
## Issue

https://gravitee.atlassian.net/browse/APIM-XXX

## Description

Migrating application general tab - 0AuthClient form

## Additional context

Screenshot:
Screenshot 2023-11-14 at 3 31 51 PM

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-zzatcykkrp.chromatic.com)

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/5848/console](https://pr.team-apim.gravitee.dev/5848/console)
      Portal: [https://pr.team-apim.gravitee.dev/5848/portal](https://pr.team-apim.gravitee.dev/5848/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/5848/api/management](https://pr.team-apim.gravitee.dev/5848/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/5848](https://pr.team-apim.gravitee.dev/5848)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/5848](https://pr.gateway-v3.team-apim.gravitee.dev/5848)

### CODE REVIEW

File: gravitee-apim-console-webui/src/entities/application/application.ts [hunk 1]
```diff
@@ -34,8 +34,14 @@ export interface Application {
...
   background?: string;
   owner?: any;
-  settings?: unknown;
+  settings?: App;
   disable_membership_notifications?: boolean;
   api_key_mode?: ApiKeyMode;
   origin?: string;
 }
+
+export interface App {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/entities/application/application.ts]:
```suggestion
export interface ApplicationSettings {
```
For the subtypes model naming we often apply this naming convention.

File: gravitee-apim-console-webui/src/management/application/details/general/general-ng/application-general.component.html [hunk 2]
```diff
@@ -96,5 +96,25 @@
     </div>
   </mat-card>
 
+  <ng-container *ngIf="!is0AuthClient">
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/application/details/general/general-ng/application-general.component.html]: we only have 2 possibilities here? I wonder if a switch wouldn't be cleaner.
or a `*ngIf="applicationType === 'oauth2' "`

File: gravitee-apim-console-webui/src/entities/application/application.ts (hunk 1 above)
Comment (jansobczynski) [reply gravitee-apim-console-webui/src/entities/application/application.ts]: Done

File: gravitee-apim-console-webui/src/management/application/details/general/general-ng/application-general.component.html (hunk 2 above)
Comment (jansobczynski) [reply gravitee-apim-console-webui/src/management/application/details/general/general-ng/application-general.component.html]: Yes, we have two types: SIMPLE, NATIVE
Done

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**